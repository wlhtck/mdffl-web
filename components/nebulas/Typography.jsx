import styled from 'styled-components';
import { oneOf, bool } from 'prop-types';
import { black, white } from './colors';

// TODO: create common margin all elements have margin: 32 0;
export const Headline = styled.h1`
  color: ${white};
  font-family: 'Raleway', sans-serif;
  margin: 0 0 32px;
  font-weight: 800;
  font-size: 36px;
`;

export const FeatureHeadline = styled.h2`
  color: ${black};
  font-family: 'Raleway', sans-serif;
  margin: 0 0 32px;
  font-weight: 700;
  font-size: 36px;
`;

export const BodyCopy = styled.p`
  text-align: ${({ textAlign }) => (textAlign)};
  color: ${({ inverted }) => (inverted ? white : black)};
  font-family: 'Lato', sans-serif;
  font-size: 24px;
  margin: 0 0 32px;
`;

BodyCopy.propTypes = {
  textAlign: oneOf(['left', 'right', 'center']),
  inverted: bool,
};

BodyCopy.defaultProps = {
  textAlign: 'left',
  inverted: false,
};
