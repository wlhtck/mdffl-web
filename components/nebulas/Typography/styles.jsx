import { black, white } from '../colors';

const marginBottom = { margin: '0 0 32px' };
const colors = { white, black };
const families = { raleway: 'Raleway', lato: 'Lato' };

/** @function fontStyles
  * Get font styles
  * @param {Object} props - The font properties.
  * @param {('left'|'right'|'center')} [props.align='left'] - css property text-align
  * @param {('black'|'white')} [props.color='black'] - css property color (text color)
  * @param {('lato'|'raleway')} [props.family='lato'] - css property color (text color)
  * @param {string} [props.size='24px'] - css property font-size
  * @param {number} [props.weight=600] - css property font-weight */

export default ({
  align = 'left', color = 'black', family = 'lato', size = '24px', weight = 600,
} = {}) => ({
  color: colors[color],
  fontFamily: `'${families[family]}', sans-serif`,
  fontWeight: weight,
  fontSize: size,
  textAlign: align,
  ...marginBottom,
});
