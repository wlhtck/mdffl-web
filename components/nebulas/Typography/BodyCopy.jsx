import ReactMarkdown from 'react-markdown';
import { oneOf, bool } from 'prop-types';
import withStyles from '../withStyles';
import styles from './styles';

const BodyCopy = withStyles(({ inverted, align }) => styles({
  align, color: (inverted ? 'white' : 'black'), size: '24px',
}), ReactMarkdown);

BodyCopy.propTypes = {
  textAlign: oneOf(['left', 'right', 'center']),
  inverted: bool,
};

BodyCopy.defaultProps = {
  textAlign: 'left',
  inverted: false,
};

export default BodyCopy;
