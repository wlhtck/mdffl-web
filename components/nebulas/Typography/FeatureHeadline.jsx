import withStyles from '../withStyles';
import styles from './styles';

const FeatureHeadline = withStyles(styles({
  family: 'raleway', weight: 700, size: '36px',
}), 'h2');

export default FeatureHeadline;
