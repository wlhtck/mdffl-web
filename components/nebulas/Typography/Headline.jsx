import { withFontStyles } from './styles';

const Headline = withFontStyles({
  color: 'white', family: 'raleway', size: '36px', weight: 800,
}, 'h1');

export default Headline;
