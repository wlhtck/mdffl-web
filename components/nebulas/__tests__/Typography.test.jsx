import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import { Headline, FeatureHeadline, BodyCopy } from '../Typography';

describe('componenets/nebulas/Typography', () => {
  describe('Headline', () => {
    it('renders a <Headline />', () => {
      expect(shallow(<Headline>Headline</Headline>)).toMatchSnapshot();
    });
  });

  describe('FeatureHeadline', () => {
    it('renders a  <FeatureHeadline />', () => {
      expect(shallow(<FeatureHeadline>Feature Headline</FeatureHeadline>)).toMatchSnapshot();
    });
  });

  describe('BodyCopy', () => {
    it('renders a <BodyCopy />', () => {
      expect(shallow(<BodyCopy>Body Copy</BodyCopy>)).toMatchSnapshot();
    });

    it('renders an inverted <BodyCopy />', () => {
      expect(shallow(<BodyCopy inverted>Body Copy</BodyCopy>)).toMatchSnapshot();
    });

    it('renders an right aligned <BodyCopy />', () => {
      expect(shallow(<BodyCopy textAlign="right">Body Copy</BodyCopy>)).toMatchSnapshot();
    });

    it('renders an centered <BodyCopy />', () => {
      expect(shallow(<BodyCopy textAlign="left">Body Copy</BodyCopy>)).toMatchSnapshot();
    });
  });
});
