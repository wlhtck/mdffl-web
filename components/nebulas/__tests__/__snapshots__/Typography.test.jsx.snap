// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componenets/nebulas/Typography BodyCopy renders a <BodyCopy /> 1`] = `
.c0 {
  text-align: left;
  color: #303133;
  font-family: 'Lato',sans-serif;
  font-size: 24px;
  margin: 0 0 32px;
}

<p
  className="c0"
>
  Body Copy
</p>
`;

exports[`componenets/nebulas/Typography BodyCopy renders an centered <BodyCopy /> 1`] = `
.c0 {
  text-align: left;
  color: #303133;
  font-family: 'Lato',sans-serif;
  font-size: 24px;
  margin: 0 0 32px;
}

<p
  className="c0"
>
  Body Copy
</p>
`;

exports[`componenets/nebulas/Typography BodyCopy renders an inverted <BodyCopy /> 1`] = `
.c0 {
  text-align: left;
  color: #FFFFFF;
  font-family: 'Lato',sans-serif;
  font-size: 24px;
  margin: 0 0 32px;
}

<p
  className="c0"
>
  Body Copy
</p>
`;

exports[`componenets/nebulas/Typography BodyCopy renders an right aligned <BodyCopy /> 1`] = `
.c0 {
  text-align: right;
  color: #303133;
  font-family: 'Lato',sans-serif;
  font-size: 24px;
  margin: 0 0 32px;
}

<p
  className="c0"
>
  Body Copy
</p>
`;

exports[`componenets/nebulas/Typography FeatureHeadline renders a  <FeatureHeadline /> 1`] = `
.c0 {
  color: #303133;
  font-family: 'Raleway',sans-serif;
  margin: 0 0 32px;
  font-weight: 700;
  font-size: 36px;
}

<h2
  className="c0"
>
  Feature Headline
</h2>
`;

exports[`componenets/nebulas/Typography Headline renders a <Headline /> 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: 'Raleway',sans-serif;
  margin: 0 0 32px;
  font-weight: 800;
  font-size: 36px;
}

<h1
  className="c0"
>
  Headline
</h1>
`;
